portability-core/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudFactory.java

```diff
@@ -94,34 +91,6 @@ public void clearJobData(String jobId) {
    */
   @Override
   public String getProjectId() {
-    return getGoogleProjectId();
-  }
-
-  /**
-   * Get project ID from environment variable and validate it is set.
-   *
-   * <p>Exposed as package private so Google classes in this package may use this directly without
-   * needing to obtain a CloudFactory instance.
-   *
-   * @return project ID
-   * @throws IllegalArgumentException if project ID is unset
-   */
-  static String getGoogleProjectId() throws IllegalArgumentException {
-    if (PROJECT_ID != null) {
-      return PROJECT_ID;
-    }
-    String tempProjectId;
-    try {
-      tempProjectId = System.getenv("GOOGLE_PROJECT_ID");
-    } catch (NullPointerException e) {
-      throw new IllegalArgumentException("Need to specify a project ID when using Google Cloud. "
-          + "This should be exposed as an environment variable by Kubernetes, see "
-          + "k8s/api-deployment.yaml");
-    }
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(tempProjectId), "Need to specify a project "
-        + "ID when using Google Cloud. This should be exposed as an environment variable by "
-        + "Kubernetes, see k8s/api-deployment.yaml");
-    PROJECT_ID = tempProjectId.toLowerCase();
-    return PROJECT_ID;
+    return projectId;
```

Comment:
Can we just get rid of CloudFactory.getProjectId if it's only used for bucket name?

Reply:
how's this? it's used for crypto key too

Reply:
it's used for crypto key too. Do you bring that up because it sounds too Google-specific to have in CloudFactory? It's meant to be general since any cloud platform should have some notion of this. Would you prefer a more generic name than project ID? (If so, what?)

Reply:
It seems to be used for forming the cryptoKeyName to pass to the key management system. 

CRYPTO_KEY_FMT_STRING seems Google specific? Can we just construct the path and use projectid or whatever, internally in the cloud specific implementation of CryptoKeyManagementSystem

Reply:
OK, now that I think about this more, the crypto key format is pretty Google-specific and the bucket name is too actually, since there is that weird requirement that bucket names have to be unique across Google projects. That's why we include project ID in the bucket name which people probably wouldn't do normally. So I agree the crypto key and bucket name conventions should be inside the Google impl's, I moved them in the latest commit